core/src/main/java/hudson/tasks/ArtifactArchiver.java

```diff
@@ -110,6 +110,12 @@
     @Nonnull
     private Boolean caseSensitive = true;
 
+    /**
+     * Indicate whether include and exclude patterns should be considered as case sensitive
+     */
+    @Nonnull
+    private Boolean followSymlinks = true;
```

Comment:
@daniel-beck, if this is false and with the readResolve() setting to true on null, does that retain old behavior for existing projects and default to not following symlinks for new ones?

We should definitely have that behavior and then I'm even more in strongly in favor of this PR.

Reply:
@jeffret-b Mostly.

This being `Boolean` and `readResolve` combine to set the the default when loading existing projects. This is definitely the correct approach to not break everyone's existing projects.

This sets the programmatic default for new instances, but the UI needs to be considered as well. https://github.com/jenkinsci/jenkins/pull/3947/files#diff-16a525c7f3ec8f3f359bc969708f920cR50 would make a different value here next to useless (modulo Pipeline snippet generator and such).

Reply:
So, change that line to `false` also, and it should work, right? That's good.

Reply:
If I understand the flow of discussion correctly, to maintain backwards compatibility for pipeline, we need to keep this as `true` and keep the new UI setting in config.jelly defaulting to false.